src/app/app.component.ts

```diff
@@ -81,8 +87,26 @@ export class MyApp {
   openPage(page) {
     // Reset the content nav to have just this page
     // we wouldn't want the back button to show in this scenario
-    this.menuCtrl.close().then(() => {  
-      this.nav.setRoot(page.component, { 'linkParam' : page.param });
+    this.menuCtrl.close().then(() => {
+      let changePageAllowed: boolean = true;
+      if(page.component === PushMessagesPage) {
+        if(!PushService.registered) {
+          changePageAllowed = false;
+          this.alert.showAlert(constants.pushAlertMessage, constants.featureNotConfigured, 
+            constants.alertButtons, constants.showDocs, constants.pushDocsUrl);
+        } 
+      } else if (page.component === AuthPage) {
+        const configKeys: string[] = Object.keys(this.auth.getConfig()).toString().split(',');
+            console.log(configKeys.length);
+            if (configKeys.length <= 1 || configKeys.indexOf('url') === -1) {
+              changePageAllowed = false;
```

Comment:
I think this is indented too far. Also could we get a comment added on the logic behind the above `if`, might not be immediately obvious to people what we're checking here.